core/arch/riscv/include/kernel/thread_arch.h

```diff
@@ -109,11 +181,46 @@ struct user_mode_ctx;
  * These flags should vary according to the privilege mode selected
  * to run OP-TEE core on (M/HS/S). For now default to S-Mode.
  */
-#define THREAD_EXCP_FOREIGN_INTR	MIP_SEIP
-#define THREAD_EXCP_NATIVE_INTR		MIP_SSIP
-#define THREAD_EXCP_ALL			(THREAD_EXCP_FOREIGN_INTR |	\
-					 THREAD_EXCP_NATIVE_INTR |	\
-					 MIP_STIP)
+
+#define CSR_XIE_SIE		BIT64(IRQ_XSOFT)
+#define CSR_XIE_TIE		BIT64(IRQ_XTIMER)
+#define CSR_XIE_EIE		BIT64(IRQ_XEXT)
+
+#define THREAD_EXCP_FOREIGN_INTR	CSR_XIE_EIE
+#define THREAD_EXCP_NATIVE_INTR		(CSR_XIE_SIE | CSR_XIE_TIE)
+#define THREAD_EXCP_ALL	(THREAD_EXCP_FOREIGN_INTR | THREAD_EXCP_NATIVE_INTR)
```

Comment:
I believe this is easier to read if you format and break the line as in the code you just removed above.